cmd/gen-go-sample/testdata/sample_unary.want

```diff
@@ -28,6 +28,14 @@ foopb "path.to/pb/foo"
 
 // [START awesome_region]
 
+// Construct a complex request object,
+// send it to the server,
+// and inspect the response.
+//
+// theX: a single-line comment for an input parameter
```

Comment:
If the comment is a single line but is really long, do you insert it as is or do you line-wrap it at some length?

Reply:
Not doing line-wrapping right now. Good point. I'll add it.

Reply:
@vchudnov-g @noahdietz Go doesn't seem to strictly enforce line length for comments (it is actually discouraged): https://github.com/golang/go/wiki/CodeReviewComments#line-length

> break lines because of the semantics of what you're writing (as a general rule) and not because of the length of the line. If you find that this produces lines that are too long, then change the names or the semantics and you'll probably get a good result.

Now that users can break lines themselves with multi-line YAML strings, I think it should be OK that we don't forcing line-wrapping in the generator. I am OK if we force newlines when the line becomes super long (e.g., 120 or 140 characters), but that to me is trying to be smart (and I don't like writing code that tries to be smart, because there are always weird edge cases).

Reply:
For the reasons you've stated, I'm cool with not doing automatic line breaks

Reply:
I know Go *allows* long lines, but I don't want us to generate uncomfortably long lines. Here's why I think  I think we should consider wrapping:
- In the other languages, we wrap lines, right?
- So then users won't be thinking about automatically inserting new lines. This is good, less for them to think about in the usual case (say they're writing a paragraph).
- If we then force them to insert line breaks for Go, then they might do it at, say, column 100. But for some other languages the word wrap happens at column 80. So then for those languages, there would be an ~80 column line, followed by a ~20 column line, followed by an ~80 column line, all part of the same paragraph. That does not look idiomatic!

I would propose something like this for Go: if the configured comment is exactly one line of less than 110 characters, output it as one line. Otherwise, wrap all the logical lines of the comment at 100 characters. (Those numbers are arbitrary, but seem reasonable for ease of reading.)

Notice that I am implicitly making a distinction between *semantic* line breaks (which the use should configure in the input when formatting matters) from *cosmetic* line breaks (which the user should not have to worry about; they're just about making the comment easier to read for the end developer).

I'm fine deferring this to a separate PR, but I would ask that we file an issue to keep track of it and see what feedback we get about comment length. (I'm also fine with implementing this now if it's easy :smile: . )

Reply:
So there are several cases here:

1) User input is a very very long one-line comment, say, 200 characters. In this case it's preferable that the generator breaks the comment into multiple lines.

2) Use input follows Go patterns (i.e., break lines because of the semantics of what they're writing), and one of the lines is slightly long, say, 111 characters. In this case I would prefer to not to break lines in the generator, because that will cause cascading effects to all following lines, which might change the line breaking that the user tried to maintain.

When it comes to cosmetic formatting, I am more inclined to have the generator to "prevent very bad things" rather than "try to make every case as good as possible". So how about we only handle case 1) here?

If we had to wrap multi-line user input as well, then we would have two choices:
a) preserves user specified line breaks, and also breaks at maximum line length. This will potentially introduce some very short lines :(

b) ignore user specified line breaks, and always breaks at maximum line length. This is just doing the exact opposite of what Go guidance prefers. :(

I just don't like either of them.

Reply:
> Use input follows Go patterns (i.e., break lines because of the semantics of what they're writing), 

The user usually won't if they're thinking about more than one language. That's what my example was about.

> and one of the lines is slightly long, say, 111 characters. In this case I would prefer to not to break lines in the generator, because that will cause cascading effects to all following lines, which might change the line breaking that the user tried to maintain.

That's why my proposal above said (emphasis added) "if the configured comment is **exactly one line** of less than 110 characters, output it as one line" 

> a) preserves user specified line breaks, and also breaks at maximum line length. This will potentially introduce some very short lines :(

That's why I proposed breaking only for single-paragraph comments. I'm guessing this will get us the 80% case.... And even if we do break for multi-paragraph comments, my proposal has some margin to prevent ultra-short lines: if the line is longer than 110 characters, break by the column 100. (Notice the difference in numbers.)

(I'd like to also be able to break for multi-line comments that don't rely on formatting, but determining that is a harder problem that is not a good use of our time right now, IMO.)

I agree, no on (b) ;-)

Reply:
Okay I think we are on the same page. I misunderstood "Otherwise" earlier, I thought that refers to "wrap all the logical lines of the comment at 100 characters when user input is either one line with more than 110 characters or multiple lines".